### ISSUE TITLE
1.21.0 now wants random.Generator.uniform(low, high) instead of also accepting uniform(high, low)

### ISSUE THREAD

**Issue by grahamgower** [numpy/numpy#19297]
Backwards compatibility has been broken as of release 1.21.0.
```python
$ python
Python 3.9.5 (default, May 24 2021, 12:50:35)
[GCC 11.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> np.__version__
'1.21.0'
>>> np.random.default_rng().uniform(10, 0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "_generator.pyx", line 944, in numpy.random._generator.Generator.uniform
  File "_common.pyx", line 578, in numpy.random._common.cont
  File "_common.pyx", line 387, in numpy.random._common.check_constraint
ValueError: high - low < 0
```

```python
$ python
Python 3.9.5 (default, May 24 2021, 12:50:35)
[GCC 11.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> np.__version__
'1.20.3'
>>> np.random.default_rng().uniform(10, 0)
6.703305006202052
```

**Comment by rkern** [comment]
That behavior was a bug (and documented as [undefined behavior since 2016](https://github.com/numpy/numpy/commit/22176f9969b526d696fa905ba74ed6c5e30f43f3)) that was fixed deliberately in #17921.

**Comment by grahamgower** [comment]
I realise that my usage was probably not correct (and it was in fact unintentional), so I've now fixed my code. But it would have been nice to see a deprecation warning or similar in the past few releases!
